source/common/grpc/async_client_manager_impl.cc

```diff
@@ -24,6 +26,16 @@ bool validateGrpcHeaderChars(absl::string_view key) {
   return true;
 }
 
+bool validateGrpcCompatibleAsciiHeaderValue(absl::string_view h_value) {
+  for (auto ch : h_value) {
+    if (ch < 0x20 || ch > 0x7e) {
+      return false;
+    }
+  }
+  return true;
+
```

Comment:
Nit: superfluous blank. If you really want you can use `return std::all_of` style here, but it's a bit of a wash.